### PR TITLE
Upgrade Spring 6/Spring Boot 3 for non-camel containers

### PR DESCRIPTION
**Pull request by msnwatson**
## What was the problem?

Out-of-date spring framework and spring boot versions were being used in containers. 

Associated tickets or Slack threads:

- #1428 

## How does this fix it?[^1]

For containers that do not make use of Camel, latest versions of the dependencies implementation are included in the build script to force upgrades to the latest Spring/Spring Boot versions.

Right now, the only containers that do not make use of Camel but do make use of Spring is `svc-lighthouse` and `svc-bip`.

I verified that latest versions of Spring dependencies are being pulled in by running the Gradle dependencies tasks for the two projects given above.

## How to test this PR
- Passing unit tests
- Successfully starting containers on local machine
- Successful deployment and healthy containers in `dev` LHDI environment
- Passing SecRel run on the two images to further confirm that outdated spring versions are not being used

[^1]: [Pull-Requests guidelines](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Pull-Requests). If PR is significant, update [Current Software State](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Current-Software-State) wiki page.
[^secrel]: To check if a PR will succeed in the SecRel workflow, [test PRs in the SecRel pipeline](https://github.com/department-of-veterans-affairs/abd-vro-internal/wiki/Secure-Release-process#to-test-prs-in-the-secrel-pipeline).

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -62,32 +64,32 @@ openApi {
...
-    dependsOn 'jacocoTestCoverageVerification'
-
-    dependsOn 'dockerPrepare'
-  }
-
-  tasks.named('spectralLint').configure {
-    dependsOn 'forkedSpringBootStop'
-  }
+//  generateOpenApiDocs.enabled = false
+//  forkedSpringBootRun.enabled = false
```
Comment (engineer-plus-plus) [inline app/build.gradle:68]: according to[ smart people](https://medium.com/codex/clean-code-comments-833e11a706dc), checking in commented code is bad

**Review by engineer-plus-plus — CHANGES_REQUESTED** [review]

**Comment by msnwatson** [comment]
https://github.com/department-of-veterans-affairs/abd-vro-internal/actions/runs/5647816828/job/15299223253 while SecRel was failing, there were no Aqua alerts caused by out-of-date spring versions after these changes

File: svc-bip-api/src/main/java/gov/va/vro/bip/service/BipApiService.java [hunk 2]
```diff
@@ -224,6 +225,10 @@ String createJwt() throws BipException {
     claims.put("iss", bipApiProps.getClaimIssuer());
     byte[] signSecretBytes = bipApiProps.getClaimSecret().getBytes(StandardCharsets.UTF_8);
     Key signingKey = new SecretKeySpec(signSecretBytes, SignatureAlgorithm.HS256.getJcaName());
+
+    String secretString = Encoders.BASE64.encode(signingKey.getEncoded());
+    System.out.println(bipApiProps.getClaimSecret());
```
Comment (yoomlam) [inline svc-bip-api/src/main/java/gov/va/vro/bip/service/BipApiService.java:230]: I would expect linters to complain about `System.out.println` calls. Why are these lines added?

File: svc-bip-api/build.gradle
Comment (yoomlam) [inline svc-bip-api/build.gradle:1]: I'm wondering if adding a `gradle.properties` like https://github.com/department-of-veterans-affairs/abd-vro/blob/4c7d466f7c903d5f0ad5efcd80161116b4ba72ce/api-gateway/gradle.properties#L5
could avoid having to specify `spring-boot-dependencies:3.1.2` and `io.jsonwebtoken:*`

File: svc-lighthouse-api/build.gradle
Comment (yoomlam) [inline svc-lighthouse-api/build.gradle:1]: Try adding a `gradle.properties` like https://github.com/department-of-veterans-affairs/abd-vro/blob/4c7d466f7c903d5f0ad5efcd80161116b4ba72ce/api-gateway/gradle.properties#L5
and removing dependencies `spring-boot-dependencies` and `io.jsonwebtoken:*`

**Review by yoomlam — COMMENTED** [review]
Left comments to try.

File: svc-bip-api/src/main/java/gov/va/vro/bip/service/BipApiService.java (hunk 2 above)
Comment (msnwatson) [reply svc-bip-api/src/main/java/gov/va/vro/bip/service/BipApiService.java:230]: whoops, i was debugging something and left this in, will rm it

File: svc-bip-api/build.gradle
Comment (msnwatson) [reply svc-bip-api/build.gradle:1]: added this which avoided me having to specify the version on `spring-boot-dependencies` but i have to include the io.jsonwebtoken explicitly or i get compile error

File: svc-lighthouse-api/build.gradle [hunk 3]
```diff
@@ -7,6 +7,7 @@ plugins {
 description = 'Automated Benefit Delivery Data Access Service'
 
 dependencies {
+  //  implementation "org.springframework.boot:spring-boot-dependencies:3.1.2"
```
Comment (yoomlam) [inline svc-lighthouse-api/build.gradle:10]:
```suggestion
```

**Review by yoomlam — APPROVED** [review]
1 suggestion

**Comment by engineer-plus-plus** [comment]
lgtm

**Review by greene858 — APPROVED** [review]
Minor question on BipException.java; but LGTM
